cyral/resource_cyral_sidecar_listener.go

```diff
@@ -0,0 +1,422 @@
+}
+type DynamoDbSettings struct {
+	ProxyMode bool `json:"proxyMode,omitempty"`
+}
+
+var ReadSidecarListenersConfig = ResourceOperationConfig{
+	Name:       "SidecarListenersResourceRead",
+	HttpMethod: http.MethodGet,
+	CreateURL: func(d *schema.ResourceData, c *client.Client) string {
+		return fmt.Sprintf("https://%s/v1/sidecars/%s/listeners/%s",
+			c.ControlPlane,
+			d.Get(SidecarIdKey).(string),
+			d.Get(ListenerIdKey).(string))
+	},
+	NewResponseData: func(_ *schema.ResourceData) ResponseData { return &ReadSidecarListenersAPIResponse{} },
+}
+
+type ReadSidecarListenersAPIResponse struct {
+	ListenerConfig *SidecarListener `json:"listenerConfig"`
+}
+type CreateListenerAPIResponse struct {
+	ListenerId string `json:"listenerId"`
+}
+
+func (c CreateListenerAPIResponse) WriteToSchema(d *schema.ResourceData) error {
+	d.SetId(marshalComposedID([]string{d.Get(SidecarIdKey).(string), c.ListenerId}, "/"))
+	_ = d.Set(ListenerIdKey, c.ListenerId)
+	return nil
+}
+
+func (data ReadSidecarListenersAPIResponse) WriteToSchema(d *schema.ResourceData) error {
+	if data.ListenerConfig != nil {
+		_ = d.Set(ListenerIdKey, data.ListenerConfig.ListenerId)
+		_ = d.Set(RepoTypesKey, data.ListenerConfig.RepoTypesAsInterface())
+		_ = d.Set(UnixListenerKey, data.ListenerConfig.UnixSettingsAsInterface())
+		_ = d.Set(TcpListenerKey, data.ListenerConfig.TcpSettingsAsInterface())
+		_ = d.Set(MultiplexedKey, data.ListenerConfig.Multiplexed)
+		_ = d.Set(S3SettingsKey, data.ListenerConfig.S3SettingsAsInterface())
+		_ = d.Set(DynamoDbSettingsKey, data.ListenerConfig.DynamoDbSettingsAsInterface())
+	}
+	return nil
+}
+func (l *SidecarListener) RepoTypesAsInterface() *[]interface{} {
+	if l.RepoTypes == nil {
+		return nil
+	}
+	result := make([]interface{}, len(l.RepoTypes))
+	for i, v := range l.RepoTypes {
+		result[i] = v
+	}
+	return &result
+}
+func (l *SidecarListener) RepoTypesFromInterface(anInterface []interface{}) {
+	repoTypes := make([]string, len(anInterface))
+	for i, v := range anInterface {
+		repoTypes[i] = v.(string)
+	}
+	l.RepoTypes = repoTypes
+}
+func (l *SidecarListener) TcpSettingsAsInterface() []interface{} {
+	if l.TcpListener != nil {
+		result := make([]interface{}, 1)
+		result[0] = map[string]interface{}{
+			HostKey: l.TcpListener.Host,
+			PortKey: l.TcpListener.Port,
+		}
+		return result
+	}
+	return nil
+}
+func (l *SidecarListener) TcpSettingsFromInterface(anInterface []interface{}) {
+	if len(anInterface) == 0 {
+		return
+	}
+	if anInterface != nil {
+		l.TcpListener = &TcpListener{
+			Host: anInterface[0].(map[string]interface{})[HostKey].(string),
+			Port: anInterface[0].(map[string]interface{})[PortKey].(int),
+		}
+	}
+	log.Printf("[DEBUG] LOLOLO: %s %d", l.TcpListener.Host, l.TcpListener.Port)
+}
+func (l *SidecarListener) UnixSettingsAsInterface() []interface{} {
+	if l.UnixListener != nil {
+		result := make([]interface{}, 1)
+		result[0] = map[string]interface{}{
+			FileKey: l.UnixListener.File,
+		}
```

Comment:
Same as above: this slice can be declared "inline" without having to first `make()` it.